lib/go/edgecontext/edgecontext.go

```diff
@@ -173,6 +175,12 @@ func New(ctx context.Context, impl *Impl, args NewArgs) (*EdgeRequestContext, er
 			ReadableID: args.RequestID,
 		}
 	}
+	if args.LocaleCode != "" {
+		request.Locale = &ecthrift.Locale{
+			LocaleCode: ecthrift.LocaleCode(args.LocaleCode),
+		}
```

Comment:
Shouldn't we apply the same regex check in here as in the python implementation?

Reply:
Makes sense to me. Added in https://github.com/reddit/edgecontext/pull/12/commits/62e3a09d745e5e3db4c0862a36a0e5cb6c19800f